src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -69,10 +73,20 @@ public int getMinWeightMagnitude() {
         return MIN_WEIGHT_MAGNITUDE;
     }
 
-    private static void runValidation(TransactionViewModel transactionViewModel, final int minWeightMagnitude) {
+    private static boolean invalidTimestamp(TransactionViewModel transactionViewModel) {
+        if (transactionViewModel.getAttachmentTimestamp() == 0) {
+            return transactionViewModel.getTimestamp() < MIN_TIMESTAMP && !transactionViewModel.getHash().equals(Hash.NULL_HASH)
+                    || transactionViewModel.getTimestamp() > (System.currentTimeMillis() / 1000) + MAX_TIMESTAMP_FUTURE;
+        }
+        return transactionViewModel.getAttachmentTimestamp() < MIN_TIMESTAMP_MS
+                || transactionViewModel.getTimestamp() > System.currentTimeMillis() + MAX_TIMESTAMP_FUTURE_MS;
```

Comment:
I see that two lines above we consider `transactionViewModel.getTimestamp()` as something that is measures in seconds. Now it is measured in ms...

Maybe you wanted to use `getAttachmentTimestamp()`??

I also recomment adding the words `seconds` to `getTimestamp` if we use seconds rather than ms.